core/trino-main/src/main/java/io/trino/execution/scheduler/ExponentialGrowthPartitionMemoryEstimator.java

```diff
@@ -117,7 +119,10 @@ void refreshNodePoolMemoryInfos()
         }
 
         @Override
-        public PartitionMemoryEstimator createPartitionMemoryEstimator(Session session, PlanFragment planFragment)
+        public PartitionMemoryEstimator createPartitionMemoryEstimator(
+                Session session,
+                PlanFragment planFragment,
+                Function<PlanFragmentId, PlanFragment> sourceFragmentLookup)
```

Comment:
@findepi with this lookup function we loose the caching of "no-memory" flag on child stages we used to have - should not be a big deal - as plans are not that huge and even assesing same plan fragment multiple times will be fine.

But maybe we can do better?

Reply:
discussed offline. let's capture this as a TODO comment for now, as this is likely not a big deal for now.
some more comprehensive profiling of the scheduler might be useful